site/topics/arithmetic/adder.rst

```diff
@@ -13,6 +13,86 @@ Half & Full Adder
 Binary Addition as a Logical Operation
 ======================================
 
+* If one wants to add two bits together, how many output bits are needed?
+* To answer this, consider the possibilities
+
+    * :math:`0_{2} + 0_{2} = 0_{2}`
+    * :math:`0_{2} + 1_{2} = 1_{2}`
+    * :math:`1_{2} + 0_{2} = 1_{2}`
+    * :math:`1_{2} + 1_{2} = 10_{2}`
+
+
+* As one can see, when adding two ``1``\s together, a carry bit is needed
+* Thus, when adding two bits together, two output bits are required
+
+* In general, when adding two :math:`n` bit numbers together, :math:`n+1` bits is sufficient to output the correct sum
+
+    * Consider that the largest number that can be represented with :math:`n` bits is :math:`2^{n} - 1`
+    * The largest sum of adding two :math:`n` bit numbers together is :math:`2(2^{n} - 1) = 2^{n+1} - 2`
+
+        * Adding the largest :math:`n` bit number to itself
+
+    * The largest number representable with :math:`n+1` bits is :math:`2^{n+1} - 1`
+    * Since :math:`2^{n+1} - 2 < 2^{n+1} - 1`, :math:`n+1` bits is sufficient
```

Comment:
This is a little long winded, and obvious. Appending a single digit increases the amount of representable digits by the factor of the number base + adding k+k=2k for any k implies 1 extra digit supports addition so long as the base >=2.  
Imo either drop the extra detail, or cram them into a `note` section so they can be glossed over.